src/liballoc/collections/linked_list.rs

```diff
@@ -878,6 +878,58 @@ impl<T> LinkedList<T> {
         unsafe { self.split_off_after_node(split_node, at) }
     }
 
+    /// Removes the element at the given index and returns it.
+    ///
+    /// This operation should compute in O(n) time.
+    ///
+    /// # Panics
+    /// Panics if at >= len
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(linked_list_remove)]
+    /// use std::collections::LinkedList;
+    ///
+    /// let mut d = LinkedList::new();
+    ///
+    /// d.push_front(1);
+    /// d.push_front(2);
+    /// d.push_front(3);
+    ///
+    /// assert_eq!(d.remove(1), 2);
+    /// assert_eq!(d.remove(0), 3);
+    /// assert_eq!(d.remove(0), 1);
+    /// ```
+    #[unstable(feature = "linked_list_remove", issue = "69210")]
+    pub fn remove(&mut self, at: usize) -> T {
+        let len = self.len();
+        assert!(at < len, "Cannot remove at a nonexistent index");
```

Comment:
```suggestion
        assert!(at < len, "Cannot remove at index outside list bounds");
```
Or something. I don't think "nonexistent index" is particularly bad, but if I saw this error message I'd probably start by thinking that I somehow managed to pass `-1` or `None` in (not that the compiler would allow it - which would confuse me for a little longer I think).

Also, just noticed this is the same as from `split_off` so maybe for consistency you can leave it :-)